kolibri2zim/scraper.py

```diff
@@ -823,8 +874,9 @@ def run(self):
                     f"FAILURE not_done={len(result.not_done)} done={len(result.done)}"
                 )
                 for future in result.done:
-                    if future.exception():
-                        raise future.exception()
+                    future_exception = future.exception()
```

Comment:
I think I prefer the former version, but that'd require a pyright inline exception

Reply:
So that's why I prefer the new version, because it does not require a pyright inline exception.

Plus pyright is relatively right here. If for some reason (there is probably none here, but who knows ...) the `exception()` method return the exception only on first call to it and then None, the old version would fail with a very mysterious bug.

Reply:
I don't like this argument. There's no way for a future with an exception to recover and loose it. If we don't trust this, there is not much we can trust anywhere in the code.

Until we use 100% properly typed code, there are still going to be a lot of areas with conditional results. I don't care for being forced one way by the type checker.
In this particular case, it's harmless because we're basically shutting down stuff but threads and future management is tricky and forcing us to use additional vars (increasing the stack and retaining objects) could be problematic elsewhere.